src/resolvers/http-and-file.ts

```diff
@@ -0,0 +1,37 @@
+import { Resolver } from '@stoplight/json-ref-resolver';
+import { parseWithPointers } from '@stoplight/yaml';
+import * as fs from 'fs';
+
+import { httpReader } from './http';
+
+// resolves files, http and https $refs, and internal $refs
+export const httpAndFileResolver = new Resolver({
+  readers: {
+    https: httpReader,
+    http: httpReader,
+    file: {
+      read(ref: any) {
+        return new Promise((resolve, reject) => {
+          const path = ref.path();
+          fs.readFile(path, 'utf8', (err, data) => {
+            if (err) reject(err);
+            resolve(data);
+          });
+        });
+      },
+    },
+  },
+
+  parseAuthorityResult: async opts => {
+    const parts = opts.targetAuthority.path().split('.');
```

Comment:
Using path.extname is likely to be a better idea here.

Reply:
A yeah I can see this causing trouble.